lib/main.dart

```diff
@@ -12,40 +16,51 @@ void main() async {
   await Firebase.initializeApp(
     options: DefaultFirebaseOptions.currentPlatform,
   );
-  runApp(
-    const ProviderScope(
-      child: App(),
-    ),
-  );
+  SystemChrome.setPreferredOrientations([DeviceOrientation.portraitUp])
+      .then((_) {
+    runApp(
+      const ProviderScope(
+        child: App(),
+      ),
+    );
+  });
 }
 
 class App extends StatelessWidget {
   const App({super.key});
 
   @override
   Widget build(BuildContext context) {
-    return MaterialApp(
-      theme: ThemeData(
-        brightness: Brightness.dark,
-        primarySwatch: Colors.blue,
-        scaffoldBackgroundColor: AppColors.backgroundColor,
-      ),
-      darkTheme: ThemeData(
-        brightness: Brightness.dark,
-        primarySwatch: Colors.blueGrey,
-        indicatorColor: Colors.blueGrey,
-        scaffoldBackgroundColor: AppColors.backgroundColor,
-      ),
-      themeMode: ThemeMode.dark,
-      debugShowCheckedModeBanner: false,
-      home: Scaffold(
-        body: Consumer(
-          builder: (context, ref, child) {
-            final isLoggedIn = ref.watch(isLoggedInProvider);
-            return isLoggedIn ? const HomeView() : const LoginView();
-          },
-        ),
-      ),
+    return Consumer(
+      builder: (context, ref, child) {
+        final configValue = ref.watch(configProvider);
+        final mainColor = configValue.value?.mainColor.toColor() ??
+            AppColors.defaultMainColor;
+
+        final isDarkModeEnabled = ref.watch(darkModeProvider);
+        final isLoggedIn = ref.watch(isLoggedInProvider);
+
+        return MaterialApp(
+          debugShowCheckedModeBanner: false,
+          theme: _buildTheme(mainColor, Brightness.light),
+          darkTheme: _buildTheme(mainColor, Brightness.dark),
+          themeMode:
+              isDarkModeEnabled == true ? ThemeMode.dark : ThemeMode.light,
```

Comment:
Trzeba sprawdzać `isEnabled == true`? czy samo `isEnabled ? x : y` nie wystarczy?

Reply:
To jest provider który asynchronicznie zaciąga więc może też nie mieć wartości oprócz zwykłego true false

Reply:
Wydaje mi się że coś się wywalało jak dałem zwykły warunek ale sprawdzę to